docs/pages/enterprise/hsm.mdx

```diff
@@ -0,0 +1,205 @@
+---
+title: Teleport HSM Support
+description: How to configure Hardware Security Modules to manage your Teleport CA private keys
+h1: Teleport HSM Support
+---
+
+Teleport can be configured to use a connected HSM to create and hold CA private
+keys.
+
+While most PKCS#11 HSMs should be supported, we test with AWS CloudHSM,
+YubiHSM2, and SoftHSM2.
+
+# Setup
+
+This guide assumes that you are familiar with setting up Teleport. HSMs need to
+be configured on all of your Auth servers. Each Auth server can be connected to
+a unique HSM, or they can all connect to the same HSM depending on your usecase.
+
+## HSM Setup
```

Comment:
To make sure the article flows from top to bottom add examples inline and use TabItem elements to let users pick the type of the HSM setup.